AK/Error.h

```diff
@@ -43,17 +48,29 @@ class Error {
     {
         return from_string_view(string);
     }
+#endif
 
     bool operator==(Error const& other) const
     {
+#ifndef KERNEL
```

Comment:
For cases like this where we have the kernel and not kernel behavior next to each other, it's easier to read if we make the preprocessor #if as a positive condition rather than a negative one

```
#if KERNEL
    return kernel_thing;
#else
    return non_kernel_thing;
#endif
```